### PR TITLE
Edit the Mir page to fix infelicities.

### PR DESCRIPTION
**Pull request by bollu**
- Remove dead reference to Mir. reflow sentence to talk about
  `Body::local_decls`
- Fix broken links to render properly.
- Add links for `Terminator`, `RETURN_PLACE`, `ProjectionElem`.

### CODE REVIEW

File: src/mir/index.md [hunk 1]
```diff
@@ -221,26 +221,26 @@ function (along with sub-instances of Mir for "promoted constants",
...
+- **Places** are identified by the enum [`Place`]. There are a few
   variants:
   - Local variables like `_1`
   - Static variables `FOO`
   - **Projections**, which are fields or other things that "project
-    out" from a base place. These are represented by the type 
-    [`Projection`].  So e.g. the place `_1.f` is a projection,
+    out" from a base place. These are represented by the type
+    [`ProjectionElem`].  So e.g. the place `_1.f` is a projection,
     with `f` being the "projection element and `_1` being the base
```
Comment (LeSeulArtichaut) [inline src/mir/index.md:241]: There's a missing `"` here that could be fixed
```suggestion
    with `f` being the "projection element" and `_1` being the base
```

**Review by LeSeulArtichaut — APPROVED** [review]
Thanks! There seem to be other references to `Mir` remaining in this file (e.g. lines 219-220), so if you have time to spare it would be great if you could fix them as well.

**Comment by bollu** [comment]
@LeSeulArtichaut done, fixed the danging ref to `Mir`

**Comment by LeSeulArtichaut** [comment]
Perfect, thanks for the PR!

**Pull request merged**